### PR TITLE
Possible Fix for Missing Info-Box After E-Mail-Confirmation

### PR DESCRIPTION
**Pull request by kergomard**
See: https://mantis.ilias.de/view.php?id=28383

Let me know if you would prefer another solution.

### CODE REVIEW

**Comment by smeyer-ilias** [comment]
I changed the assignment, since it these changes affect Services/Initialisation

File: Services/Init/classes/class.ilStartUpGUI.php [hunk 1]
```diff
@@ -271,6 +278,24 @@ public static function printToGlobalTemplate($tpl)
         $gtpl->setContent($tpl->get());
         $gtpl->printToStdout("DEFAULT", false, true);
     }
+    
+    protected function retrieveMessagesFromSession() : array
+    {
+        $messages = [];
+        if (isset($_SESSION[ilGlobalTemplate::MESSAGE_TYPE_FAILURE])) {
+            $messages[ilGlobalTemplate::MESSAGE_TYPE_FAILURE] = $_SESSION[ilGlobalTemplate::MESSAGE_TYPE_FAILURE];
```
Comment (smeyer-ilias) [inline Services/Init/classes/class.ilStartUpGUI.php]: Please avoid the direct access to $_SESISON and use ilSession.

**Review by smeyer-ilias — CHANGES_REQUESTED** [review]
Thanks for fixing this.

File: Services/Init/classes/class.ilStartUpGUI.php (hunk 1 above)
Comment (kergomard) [reply Services/Init/classes/class.ilStartUpGUI.php]: Done

**Comment by mjansenDatabay** [comment]
IMO this is only an issue for `release_7`. This seems to be fixed in the meantime for ILIAS >= 8.x.

**Comment by mjansenDatabay** [comment]
This solution differs from the (multiple) fixes in ILIAS >= 8.x (I am still not sure which appraoch I prefer :-)), but I'll accept this for `release_7`.

**Comment by mjansenDatabay** [comment]
@kergomard Could you please resolve the conflicts and rebase/force-push your branch afterwards?

**Comment by kergomard** [comment]
Thank you very much @mjansenDatabay for having a look at this. I rebased and force-pushed.

**Comment by iszmais** [comment]
@kergomard Thanks for that improvment!
coincidentally i worked on the same problem yesterday i correlation of another Bug in ILIAS 8.

First comes first serves. I would revert my solution.
Could you provide this fix for release_8 and trunk, too?

Greetings,
@iszmais

**Comment by kergomard** [comment]
@iszmais , I'm now unsure: Do we need parts of this for trunk or not? I can provide a separate PR, if needed, no problem.

**Comment by iszmais** [comment]
@kergomard As far as i know the part wich provides the persistence of messages (so that they will not will be lost after deleting the session) is needed for 8 and trunk, therefore the first 3 additions of your PR.

I cant say anything for the other parts.

**Comment by iszmais** [comment]
But as i would assume and , as noone seem to have a veto on that, that this should be fully Cerry picked in ILIAS 8 and TRUNK.

**Pull request merged**